Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -2382,6 +2384,11 @@ Bytecode::CodeGenerationErrorOr<void> TryStatement::generate_bytecode(Bytecode::
     }
     if (m_finalizer)
         generator.end_boundary(Bytecode::Generator::BlockBoundaryType::ReturnToFinally);
+    if (m_handler) {
+        if (!m_finalizer)
+            unwind_context.emplace(generator, OptionalNone());
+        unwind_context->set_handler(handler_target.value());
+    }
```

Comment:
this is pretty ugly, you're relying on the fact that the constructor only takes the finalizer to make sure the handler block only has the finalizer set, and not itself as its handler, and then you're setting the handler midway to make sure it _is_ set on on the try block
That was pretty confusing and uninituitive to me when I first tried to review this.
If you can think of a way to simplify it, that would be great, otherwise some comments to explain this might be good enough instead.